File: api/src/main/java/net/impactdev/gts/api/listings/ui/EntryUI.java
package net.impactdev.gts.api.listings.ui;

import net.impactdev.impactor.api.platform.players.PlatformPlayer;
import net.impactdev.impactor.api.ui.containers.icons.Icon;

import java.util.Optional;

public interface EntryUI<E> {

	Optional<E> getChosenOption();

	void open(PlatformPlayer user);

	Icon<?> generateWaitingIcon(boolean auction);

	Icon<?> generateConfirmIcon();

	Icon<?> createNoneChosenIcon();

	Icon<?> createChosenIcon();

	Icon<?> createPriceIcon();

	Icon<?> createTimeIcon();

	void style(boolean selected);

}

File: api/src/main/kotlin/net/impactdev/gts/api/listings/ui/EntryUI.kt
package net.impactdev.gts.api.listings.ui

import java.util.*

interface EntryUI<P, E, I> {
    val chosenOption: Optional<E>?
    fun open(user: P)
    fun generateWaitingIcon(auction: Boolean): I
    fun generateConfirmIcon(): I
    fun createNoneChosenIcon(): I
    fun createChosenIcon(): I
    fun createPriceIcon(): I
    fun createTimeIcon(): I
    fun style(selected: Boolean)
}